### ISSUE TITLE
Fails to parse valid toml configuration

### ISSUE THREAD

**Issue by chavacava** [mgechev/revive#470]
**Describe the bug**
`revive` fails with an error when executed with certain configurations, for example:

```toml
[rule.add-constant]
  arguments = [{maxLitCount = "3"}]
```

The output error message is from the toml parsing library:
```
cannot parse the config file: (0, 0): Can't convert [maxLitCount = "3"]([]*toml.Tree) to trees
```

**To Reproduce**
1. I updated revive `go get -u github.com/mgechev/revive`
2. I run it with the following flags & configuration file:

```shell
revive -config valid.toml
```
[rule.add-constant]
  arguments = [{maxLitCount = "3"}]
```toml

```

**Expected behavior**
`revive` executes without error

**Additional context**
The problem seem to be introduced by PR #465 that switches TOML parsing library.
The new library, `github.com/pelletier/go-toml`, seems to have problems parsing arrays of objects like the one used in the configuration above.
The library proposes a toml linter utility `tomll` (that also rewrites the input file(!?)). When linting the config file with `tomll` it returns no errors and rewrites the file content to:
```toml
[rule]

  [rule.add-constant]

    [[rule.add-constant.arguments]]
      maxLitCount = "3"
```
The parsing library also fails when trying to parse this new version of the configuration.
